src/main/java/org/folio/service/kbcredentials/UserKbCredentialsServiceImpl.java

```diff
@@ -35,23 +37,18 @@
   private final AssignedUserRepository assignedUserRepository;
   private final Converter<DbKbCredentials, KbCredentials> credentialsFromDBConverter;
 
-
-  public UserKbCredentialsServiceImpl(KbCredentialsRepository credentialsRepository,
-                                      AssignedUserRepository assignedUserRepository,
-                                      Converter<DbKbCredentials, KbCredentials> credentialsFromDBConverter) {
-    this.credentialsRepository = credentialsRepository;
-    this.assignedUserRepository = assignedUserRepository;
-    this.credentialsFromDBConverter = credentialsFromDBConverter;
-  }
+  private final VertxCache<String, KbCredentials> cache;
```

Comment:
Let's think carefully if it's possible to avoid using cache. The problem might be with excessive number of calls to the service but not with the long response.